### ISSUE TITLE
Adding Dice Coefficient, Tversky Loss, etc.

### ISSUE THREAD

**Issue by DavidLandup0** [keras-team/keras-cv#894]
**Short Description**

With the addition of DeepLab by @qlzh727, it might be a good idea to add relevant metrics and loss functions that are commonly used for segnets. If we do Dice Loss, it might be better to generalize to Tversky Loss as a parent class, and then override for other losses like Dice.

**Other Information**

I'd gladly work on any of these implementations as well!

**Comment by innat** [comment]
@DavidLandup0 
Few mons ago, I made two PR regarding segmentation loss and they're inactive now. If you've time, you're most welcome to take them from here.

https://github.com/keras-team/keras-cv/pull/371 https://github.com/keras-team/keras-cv/pull/449

cc. @blackhat-coder : https://github.com/keras-team/keras-cv/pull/414

**Comment by DavidLandup0** [comment]
Whoops, my bad for not seeing them. Thanks!
